common/element_handle.go

```diff
@@ -296,85 +300,88 @@ func (h *ElementHandle) click(p *Position, opts *MouseClickOptions) error {
 }
 
 func (h *ElementHandle) clickablePoint() (*Position, error) {
-	var quads []dom.Quad
-	var layoutViewport *cdppage.LayoutViewport
-	var err error
-
-	action := dom.GetContentQuads().
-		WithObjectID(h.remoteObject.ObjectID)
-	if quads, err = action.Do(cdp.WithExecutor(h.ctx, h.session)); err != nil {
-		return nil, fmt.Errorf("cannot request node content quads %T: %w", action, err)
+	var (
+		quads []dom.Quad
+		err   error
+	)
+	getContentQuads := dom.GetContentQuads().WithObjectID(h.remoteObject.ObjectID)
+	if quads, err = getContentQuads.Do(cdp.WithExecutor(h.ctx, h.session)); err != nil {
+		return nil, fmt.Errorf("cannot request node content quads %T: %w", getContentQuads, err)
 	}
 	if len(quads) == 0 {
 		return nil, fmt.Errorf("node is either not visible or not an HTMLElement: %w", err)
 	}
 
-	action2 := cdppage.GetLayoutMetrics()
-	if layoutViewport, _, _, _, _, _, err = action2.Do(cdp.WithExecutor(h.ctx, h.session)); err != nil {
-		return nil, fmt.Errorf("cannot get page layout metrics %T: %w", action, err)
+	// Filter out quads that have too small area to click into.
+	var layoutViewport *cdppage.LayoutViewport
+	getLayoutMetrics := cdppage.GetLayoutMetrics()
+	if _, _, _, layoutViewport, _, _, err = getLayoutMetrics.Do(cdp.WithExecutor(h.ctx, h.session)); err != nil {
+		return nil, fmt.Errorf("cannot get page layout metrics %T: %w", getLayoutMetrics, err)
 	}
 
-	// Filter out quads that have too small area to click into.
-	clientWidth := layoutViewport.ClientWidth
-	clientHeight := layoutViewport.ClientHeight
+	return filterQuads(layoutViewport.ClientWidth, layoutViewport.ClientHeight, quads)
+}
+
+func filterQuads(viewportWidth, viewportHeight int64, quads []dom.Quad) (*Position, error) {
 	var filteredQuads []dom.Quad
 	for _, q := range quads {
 		nq := q
 		for i := 0; i < len(q); i += 2 {
-			nq[i] = math.Min(math.Max(q[i], 0), float64(clientWidth))
-			nq[i+1] = math.Min(math.Max(q[i+1], 0), float64(clientHeight))
+			nq[i] = math.Min(math.Max(q[i], 0), float64(viewportWidth))
+			nq[i+1] = math.Min(math.Max(q[i+1], 0), float64(viewportHeight))
```

Comment:
Do you know if this actually does anything? `nq` is never used after this. Does it also modify `q`?

Reply:
Yep... They are different slices with the same backing array. Good catch, nevertheless. Eagle eyes you have :)

That algorithm makes the points within the viewport and positive.